onnxruntime/core/providers/cpu/tensor/gather_elements.h

```diff
@@ -28,4 +28,9 @@ class GatherElements final : public OpKernel {
   int64_t axis_;
 };
 
+// holds common checks for the CPU and CUDA GatherElements kernel
+Status ValidateInputShapes(const TensorShape& input_data_shape,
+                           const TensorShape& indices_shape,
```

Comment:
nit: Would be nicer as a static method in GatherElements so it has a meaningful scope (onnxruntime::ValidateInputShapes(...) seems far too global)

Reply:
Fixed.